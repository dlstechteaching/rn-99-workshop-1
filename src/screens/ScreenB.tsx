import { StyleSheet, View, TouchableOpacity, Image, Text, ScrollView  } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { StatusBar } from 'expo-status-bar';
import images from '../assets';
import { useNavigation } from '@react-navigation/native';
import { MyCard } from '../components/common/MyCard';
import { dataForBScreen } from './hardcoded_data';

export const ScreenB : React.VFC = () => {
	const { goBack } = useNavigation();

	const cards = dataForBScreen;    
    
	return (
		<View style={styles.container}>
			<StatusBar />
			<View style={styles.headerContainer}>
				<View>
					<Text style={styles.title}>
                   My Tasks
					</Text>
					<Text style={styles.subtitle}>
                    You have 4 tasks today
					</Text>
				</View>
				<Image source={images.profile} />
			</View>
			<ScrollView 
				horizontal
				style={{overflow: 'visible'}}
				showsVerticalScrollIndicator={false}
				contentContainerStyle={styles.contentContainer}>
				{
					cards.map((card: any) => <MyCard key={card.uid} {...card}/>)
				}
			</ScrollView>
			<TouchableOpacity onPress={goBack} style={styles.button}>
				<Ionicons name={'ios-add-sharp'} size={18} color="green" />
			</TouchableOpacity>
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		flex: 1,
		flexDirection: 'column',
		padding: 30,
		width: '100%',
		backgroundColor: 'rgb(88, 191, 193)',
		justifyContent: 'space-between'
	},
	headerContainer: { 
		marginTop: 120,
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',
		marginBottom: 40
	},
	title : {
		fontSize: 32,
		color: 'white'
	},
	subtitle: {
		fontSize: 16,
		color: 'white',
		opacity: 0.7
	},
	contentContainer : {
		overflow: 'visible',
		height: '70%'
	},
	button: {
		alignSelf: 'center',
		marginBottom: 30,
		borderRadius: 35,
		flexShrink: 1,
		backgroundColor: 'rgb(236, 248, 248)',
		height: 55,
		width: 55,
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
		shadowColor: '#000',
		shadowOffset: { width: 1, height: 1 },
		shadowOpacity:  0.4,
		shadowRadius: 3,
		elevation: 5,
	},

});