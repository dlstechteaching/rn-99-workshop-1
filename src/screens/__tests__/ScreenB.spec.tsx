import { useNavigation } from '@react-navigation/native';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { ScreenB } from '../ScreenB';

jest.mock('@react-navigation/native');

describe('ScreenB tests', () => {
	let component: ShallowWrapper;
	const navigationMock = {
		navigate: jest.fn(),
	};

	beforeEach(() => {
		(useNavigation as jest.Mock).mockReturnValue(navigationMock);
	});

	it('should render properly', () => {
		givenComponent();
		thenItRendersProperly();
	});

	function givenComponent() {
		component = shallow(<ScreenB />);
	}

	function thenItRendersProperly() {
		expect(component).toMatchSnapshot();
	}
});
