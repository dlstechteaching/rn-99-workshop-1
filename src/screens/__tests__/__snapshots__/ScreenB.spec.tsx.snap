// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenB tests should render properly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(88, 191, 193)",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "padding": 30,
      "width": "100%",
    }
  }
>
  <ExpoStatusBar />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 40,
        "marginTop": 120,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 32,
          }
        }
      >
        My Tasks
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "opacity": 0.7,
          }
        }
      >
        You have 4 tasks today
      </Text>
    </View>
    <Image
      source={1}
    />
  </View>
  <ScrollView
    contentContainerStyle={
      Object {
        "height": "70%",
        "overflow": "visible",
      }
    }
    horizontal={true}
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "overflow": "visible",
      }
    }
  >
    <MyCard
      badges={
        Array [
          "#Design",
          "#Visual",
          "#SPRINT 2",
        ]
      }
      comments={
        Array [
          "test",
          "test",
          "test",
        ]
      }
      desc="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque ultricies commodo erat, vel laoreet mi porta eget. Donec blandit orci mollis ex auctor, in hendrerit nunc pretium"
      key="0"
      title="Create a new landing page Hero image"
      type="DESIGN"
      uid={0}
    />
    <MyCard
      badges={
        Array [
          "#Design",
          "#Visual",
          "#SPRINT 2",
        ]
      }
      comments={
        Array [
          "test",
          "test",
          "test",
        ]
      }
      desc="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque ultricies commodo erat, vel laoreet mi porta eget. Donec blandit orci mollis ex auctor, in hendrerit nunc pretium"
      key="1"
      title="Create a new landing page Hero image"
      type="DESIGN"
      uid={1}
    />
    <MyCard
      badges={
        Array [
          "#Design",
          "#Visual",
          "#SPRINT 2",
        ]
      }
      comments={
        Array [
          "test",
          "test",
          "test",
        ]
      }
      desc="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque ultricies commodo erat, vel laoreet mi porta eget. Donec blandit orci mollis ex auctor, in hendrerit nunc pretium"
      key="2"
      title="Create a new landing page Hero image"
      type="DESIGN"
      uid={2}
    />
  </ScrollView>
  <ForwardRef
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "rgb(236, 248, 248)",
        "borderRadius": 35,
        "display": "flex",
        "elevation": 5,
        "flexShrink": 1,
        "height": 55,
        "justifyContent": "center",
        "marginBottom": 30,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 0.4,
        "shadowRadius": 3,
        "width": 55,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="green"
      name="ios-add-sharp"
      size={18}
    />
  </ForwardRef>
</View>
`;
